design/one-pager-resource-status.md

```diff
@@ -0,0 +1,135 @@
+represent high level _"stop-light"_ conditions.
+
+The high level conditions would be:
+- Online
+  > _When a resource instance is fully available for use with no errors or failures_
+- Offline
+  > _When a resource instance is **not** available for use, potentially with many errors and failures_
+- Warning
+  > _When a resource instance is **Online** but may have some errors or failures that do not affect its "Readiness"_
+- Unknown
+  > _This would be returned if the conditions were annotated but we cannot verify a known state based on the given rules._
+  > _(Essentially a default case in a switch block)_
+
+## Design
+
+In contrast to the design [here](https://github.com/crossplaneio/crossplane/blob/master/design/one-pager-stack-status-spec.md),
+this design is meant to handle combinations of arbitrary statuses to determine a high level "overall" status.
+
+The following examples are meant to be added to a `ui-schema.yaml` file. The `resourceStatus` key should be added as a root key in that file.
+
+<a name="example-1"></a>
+
+```yaml
+# Example 1
+resourceStatus:
+  paths: .status.conditionedStatus.conditions
+  states:
+    Online:
+    - Ready: 'True'
+      Synchronized: 'True'
+    Offline:
+    - Ready: 'False'
+      Synchronized: 'False'
+    Warning:
+    - Ready: 'False'
+      Synchronized: 'True'
+      SyncFailure: 'True'
+```
+
+<a name="example-2"></a>
+
+```yaml
+# Example 2
+resourceStatus:
+  paths:
+    Ready: .status.conditionedStatus.conditions[?(@.type=='Ready')]
+    Synchronized: .status.conditionedStatus.conditions[?(@.type=='Synchronized')]
+    SyncFailure: # Example 2a
+      status: .status.someRandomStatuses.SyncFailure.value
+      message: .status.someRandomStatuses.SyncFailure.text
+      lastTransitionTime: .status.someRandomStatuses.SyncFailure.updatedTime
+  states:
+    Online:
+    - Ready: True
+      Synchronized: True
+      SyncFailure: False
+    - Ready: True
+      Synchronized: True
+      SyncFailure: not 'True'
+    Offline:
+    - Ready: False
+      Synchronized: False
+    - Ready: True
+      Synchronized: not 'True'
+      SyncFailure: True
+    Warning:
+    - Ready: 'False'
+      Synchronized: 'True'
+      SyncFailure: 'True'
+```
+
+There are two main keys within `resourceStatus`. The first is `paths` which can be a string or map of key/value pairs. Additionally the key/value pairs can be set as <string>: <string> **OR** <string>: <object>.
+
+To expand on how to set the `paths` key a bit:
+
+- If the **states** are using status types that can all be found within a single conditions array that follow the object
+structure of a standard [PodCondition](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.17/#podcondition-v1-core),
+then you can just set the `paths` key to the JSONPath of that conditions array. (_[Example 1](#example-1)_)
+
+- If the states are using status types from various or arbitrary locations, then you can individually set each path as a
```

Comment:
the marshalling/unmarshalling for this might be a little odd since we're saying that the `paths` field can either be an array or it could be a map/dictionary.  golang doesn't handle this super smoothly that I know of (at least the code consuming this data will need to be aware of the various types it could be and do some checks to know how to proceed), but maybe it works better in javascript/typescript?

Reply:
Referring to resolution above: https://github.com/crossplaneio/crossplane/pull/1144#discussion_r363055081

We also hashed this out in person but we came to the consensus that this is very manageable in js/ts so we should be ok for now.